packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -1,35 +1,351 @@
+        </div>`
+      )
+      .join();
+
+  const today = new Date();
+  const producerText = getTenantText(
+    producer.name,
+    producer.externalId.origin,
+    producer.externalId.value
+  );
+
+  const consumerText = getTenantText(
+    consumer.name,
+    consumer.externalId.origin,
+    consumer.externalId.value
+  );
+  const [submitter, submissionTimestamp] = await getSubmissionInfo(
+    selfcareId,
+    seed
+  );
+  const [activator, activationTimestamp] = await getActivationInfo(
+    selfcareId,
+    seed
+  );
+
+  const { certified, declared, verified } = await getAttributeInvolved(
+    consumer,
+    seed,
+    attributeQuery
+  );
+
+  return {
+    todayDate: dateAtRomeZone(today),
+    todayTime: timeAtRomeZone(today),
+    agreementId: agreement.id,
+    submitter,
+    submissionDate: dateAtRomeZone(submissionTimestamp),
+    submissionTime: timeAtRomeZone(submissionTimestamp),
+    activator,
+    activationDate: dateAtRomeZone(activationTimestamp),
+    activationTime: timeAtRomeZone(activationTimestamp),
+    eServiceName: eservice.name,
+    producerText,
+    consumerText,
+    certifiedAttributes: getCertifiedAttributeHtml(certified),
+    declaredAttributes: getDeclaredAttributeHtml(declared),
+    verifiedAttributes: getVerifiedAttributeHtml(verified),
+  };
+};
+
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export const contractBuilder = (
-  selfcareId: SelfcareId,
   attributeQuery: AttributeQuery,
-  storeFile: FileManager["storeBytes"],
+  pdfGenerator: PDFGenerator,
+  fileManager: FileManager,
+  config: AgreementProcessConfig,
   logger: Logger
-) => ({
-  createContract: async (
-    agreement: Agreement,
-    eservice: EService,
-    consumer: Tenant,
-    producer: Tenant,
-    seed: UpdateAgreementSeed
-  ): Promise<ApiAgreementDocumentSeed> =>
-    await pdfGenerator.createDocumentSeed(
-      selfcareId,
-      agreement,
-      eservice,
-      consumer,
-      producer,
-      seed,
-      attributeQuery,
-      storeFile,
-      logger
-    ),
-});
+) => {
+  const filename = fileURLToPath(import.meta.url);
+  const dirname = path.dirname(filename);
+
+  return {
+    createContract: async (
+      selfcareId: SelfcareId,
+      agreement: Agreement,
+      eservice: EService,
+      consumer: Tenant,
+      producer: Tenant,
+      seed: UpdateAgreementSeed
+    ): Promise<ApiAgreementDocumentSeed> => {
+      const templateFilePath = path.resolve(
+        dirname,
+        "..",
+        "resources/templates/documents",
+        "agreementContractTemplate.html"
+      );
+
+      const pdfPayload = await getPdfPayload(
+        selfcareId,
+        agreement,
+        eservice,
+        consumer,
+        producer,
+        seed,
+        attributeQuery
+      );
+
+      const documentId = generateId<AgreementDocumentId>();
+      const documentName = createAgreementDocumentName(
+        agreement.consumerId,
+        agreement.producerId
+      );
```

Comment:
I would move these after the `pdfGenerator.generate` since we will need them after it